src/main/java/duke/tasklist/TaskManager.java

```diff
@@ -0,0 +1,235 @@
+            throw new DukeException(INCORRECT_DONE_COMMAND);
+        }
+    }
+
+    /**
+     * Performs search function
+     *
+     * @param taskList Task type arraylist to store all the tasks entered by the user
+     * @param userInput takes input from the keyboard
+     * @throws DukeException if task containing keyword has been found but cannot be added into the search list
+     */
+    public static void find(ArrayList<Task> taskList, String userInput) throws DukeException {
+        String searchedTask = userInput.replace("find ", " ").trim();
+        ArrayList<Task> searchList = new ArrayList<>();
+        int taskListIndex = 0;
+        int searchListIndex = 0;
+        boolean isFound = isTaskFound(taskList, searchedTask);
+
+        addTaskToSearchList(taskList, searchedTask, searchList, taskListIndex, searchListIndex);
+        printValidSearchMessage(isFound);
+        printSearchList(searchList);
+    }
+
+    /**
+     * Adds tasks that contain a specified keyword by the user in the task list into the search list.
+     *
+     * @param taskList Task type arraylist to store all the tasks entered by the user
+     * @param searchedTask takes keyword from the keyboard
+     * @param searchList Task type arraylist to store all the tasks containing keywords entered by the user
+     * @param searchListIndex index where task should be added in search list
+     * @param taskListIndex index of task in task list to be checked for whether it contains keywords entered by user
+     *
+     * @throws DukeException if task containing keyword has been found but cannot be added into the search list
+     */
+    private static void addTaskToSearchList(ArrayList<Task> taskList, String searchedTask, ArrayList<Task> searchList, int taskListIndex, int searchListIndex) throws DukeException {
+        for (Task task : taskList) {
+            if ((task.getDescription().contains(searchedTask))) {
+                String taskType = taskList.get(taskListIndex).getTaskType();
+                switch (taskType) {
+                case "E":
+                    addTaskAsEvent(searchList, "event " + task.description + " /at" + task.getDeadline(), true);
+                    if(task.getStatusIcon().equals("X")){
+                       searchList.get(searchListIndex).updateIsDone();
+                    }
+                    break;
+                case "D":
+                    addTaskAsDeadline(searchList,"deadline " + task.description + " /by" + task.getDeadline(), true);
+                    if(task.getStatusIcon().equals("X")){
+                        searchList.get(searchListIndex).updateIsDone();
+                    }
+                    break;
+                case "T":
+                    addTaskAsToDo(searchList, task.description, true);
+                    if(task.getStatusIcon().equals("X")){
+                        searchList.get(searchListIndex).updateIsDone();
+                    }
+                    break;
```

Comment:
aren't all of them treated the same way when being searched? do you really need this `switch ... case`?